### PR TITLE
Create the Joomla StackPack

### PR DESCRIPTION
**Pull request by jeckodevelopment**
Credit @renekreijveld

### CODE REVIEW

File: packs/joomla.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/**
+ * @license Copyright 2019 The Lighthouse Authors. All Rights Reserved.
```
Comment (housseindjirdeh) [inline packs/joomla.js:2]:
```suggestion
 * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
```

**Comment by housseindjirdeh** [comment]
@jeckodevelopment Only thing that's left to do is add a Joomla detection to the [library detector](https://github.com/johnmichel/Library-Detector-for-Chrome/blob/master/library/libraries.js) used within Lighthouse 🎉

**Comment by jeckodevelopment** [comment]
Thank you @housseindjirdeh !

**Comment by jeckodevelopment** [comment]
@housseindjirdeh added the code to detect Joomla.
For your review

**Comment by housseindjirdeh** [comment]
Thanks @jeckodevelopment :)

File: packs/joomla.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  uses_webp_images: 'Consider using a [plugin](https://extensions.joomla.org/instant-search/?jed_live%5Bquery%5D=webp) or service that will automatically convert your uploaded images to the optimal formats.',
+  /** Additional description of a Lighthouse audit that tells the user how they can improve performance by lazy loading images that are initially offscreen in the context of the Joomla CMS platform. This is displayed after a user expands the section to see more. No character length limits. Links in (parenthesis) become link texts to additional documentation. */
+  offscreen_images: 'Install a [lazy-load Joomla plugin](https://extensions.joomla.org/instant-search/?jed_live%5Bquery%5D=lazy%20loading) that provides the ability to defer any offscreen images, or switch to a template that provides that functionality. Also consider using [an AMP plugin](https://extensions.joomla.org/instant-search/?jed_live%5Bquery%5D=amp).',
```
Comment (igrigorik) [inline packs/joomla.js:20]: With the WIP lazyload work in progress.. fingers crossed, this should become mostly irrelevant, correct?

Comment (zero-24) [reply packs/joomla.js:20]: Correct. But even that search gives you the backported plugin for 3.x as first result. So for 3.x i think this is good we might have to check here again when 4.x is finally released.

**Comment by connorjclark** [comment]
joomla was never added to lighthouse core. should it have been? @housseindjirdeh @jeckodevelopment 

we only use these stacks: https://github.com/GoogleChrome/lighthouse/blob/74bb627289d1ebd19d7b51fcd1209bf47e4bdfdf/lighthouse-core/lib/stack-packs.js#L29 (maybe we shouldn't have an allow list ...)

**Comment by jeckodevelopment** [comment]
Imho it should be added, but i'd leave the decision to @housseindjirdeh

**Comment by connorjclark** [comment]
We are translating these strings. Are UI elements in Joomla i18n'd?

cc @renekreijveld

**Pull request merged**